### DISCUSSION TITLE
Arcus Logging - Wrong status code in app Insigths

### DISCUSSION THREAD

**Discussion by JLLudovico** [arcus-azure/arcus.webapi#208] [Q&A] [answered]
Hi all. "Quick" question regarding arcus:
we are using arcus logging at a project and we have the following middleware config in a web app api startup.cs
-    **app.UseExceptionHandling()**
-    app.UseHttpCorrelation();
-    app.UseRouting();
-    **app.UseRequestTracking();**
    
With this config, when we have an exception, the request is logged in app insights with an HTTP 200 status code

When we change the config to:
-  **app.UseRequestTracking();**
-   **app.UseExceptionHandling();**
-  app.UseHttpCorrelation();
-  app.UseRouting();
        
When we have the exception app Insights logs the request with a HTTP 500 status code (which is correct).

But according to the [arcus Logging documentation](https://github.com/arcus-azure/arcus.webapi/blob/master/docs/features/logging.md#logging-unhandled-exceptions) it says that the .UseExceptionHandling() should be the first middleware to use.

Is this a bug or probably some misconfiguration in my startup.cs? thx

**Comment by tomkerkhove** [comment, marked as answer]
This looks like a documentation bug where the request tracking should be the first, thanks for letting us know!
Can you pick this up @stijnmoreels?

**Reply by stijnmoreels** [reply]
Hmm, yeah, our templates use the exception handling first (as would my choice, actually). I'll verify with some request tracking tests maybe on our templates. 
The problem here, probably, is that when an unhandled exception is raised, the exception handling middleware doesn't have time to set the status code to 500, so the request tracking middleware takes the current status code (which is 200).

**Reply by stijnmoreels** [reply]
So, in that regard, we could set the request tracking first, at the condition it's having it's own exception handling when the tracking fails.

**Reply by stijnmoreels** [reply]
We updated this @JLLudovico , in the docs of the web API and in our project templates. Thanks again!
